tree/treeplayer/test/data.h

```diff
@@ -2,14 +2,39 @@
 #define TTREEREADER_LEAF_TEST_DATA
 
 #include <vector>
-struct Data {
-   unsigned int fUSize;
-   int fSize;
-   double* fArray; //[fSize]
-   float* fUArray; //[fUSize]
-   std::vector<double> fVec;
-   Double32_t fDouble32;
-   Float16_t fFloat16;
+class Data {
+   unsigned int fUSize{2};
+   int fSize{4};
+   double *fArray{nullptr}; //[fSize]
+   float *fUArray{nullptr}; //[fUSize]
+   std::vector<double> fVec{17., 18., 19., 20., 21., 22.};
+   Double32_t fDouble32{17.};
+   Float16_t fFloat16{44.};
+
+public:
+   explicit Data(bool /*justToHaveAUserDefinedConstructor*/)
+      : fArray(new double[4]{12., 13., 14., 15.}), fUArray(new float[2]{42., 43.})
+   {
```

Comment:
```suggestion
   unsigned int fUSize{0};
   int fSize{0};
   double *fArray{nullptr}; //[fSize]
   float *fUArray{nullptr}; //[fUSize]
   std::vector<double> fVec{};
   Double32_t fDouble32{0};
   Float16_t fFloat16{0};

public:
   explicit Data(bool /*justToHaveAUserDefinedConstructor*/)
      : fUSize(2), fSize(4), fArray(new double[4]{12., 13., 14., 15.}), fUArray(new float[2]{42., 43.}), fVec({17., 18., 19., 20., 21., 22.}), fDouble32(17), fFloat16(44.)
   {
```
(alternative we can use a init function to set the non-zero values).